packages/serverpod_client/lib/src/serverpod_client_shared.dart

```diff
@@ -200,6 +227,7 @@ abstract class ServerpodClientShared extends EndpointCaller {
         'host must end with a slash, eg: https://example.com/');
     assert(host.startsWith('http://') || host.startsWith('https://'),
         'host must include protocol, eg: https://example.com/');
+    _requestDelegate = ServerpodClientRequestDelegateImpl(this);
```

Comment:
Suggestion: Only pass in fields that are used.

When we have god classases such as `ServerpodClientShared` it is nice to limit what is exposed of it when instantiating other classses. This makes it very clear what is actually used. Since a reference to the `ServerpodCientShared` instance is not needed, we might as well simply pass the what we need:
- `securityContext`
- `connectionTimeout`
- `serializationManager`

An added benifit of this change is that we don't need to store `securityContext` in the shared class anymore.